app/bundles/EmailBundle/Swiftmailer/Transport/MomentumTransport.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\Swiftmailer\Transport;
+
+use Mautic\EmailBundle\Swiftmailer\Momentum\Callback\MomentumCallback;
+use Mautic\EmailBundle\Swiftmailer\Momentum\Facade\MomentumFacadeInterface;
+use Swift_Events_EventListener;
+use Swift_Mime_Message;
+use Symfony\Component\HttpFoundation\Request;
+
+class MomentumTransport implements \Swift_Transport, TokenTransportInterface, CallbackTransportInterface
+{
+    /**
+     * @var MomentumFacadeInterface
+     */
+    private $momentumFacade;
+
+    /**
+     * @var \Swift_Events_SimpleEventDispatcher
+     */
+    private $swiftEventDispatcher;
+
+    /**
+     * @var bool
+     */
+    private $started = false;
+
+    /**
+     * @var MomentumCallback
+     */
+    private $momentumCallback;
+
+    /**
+     * MomentumTransport constructor.
+     *
+     * @param MomentumCallback        $momentumCallback
+     * @param MomentumFacadeInterface $momentumFacade
+     */
+    public function __construct(
+        MomentumCallback $momentumCallback,
+        MomentumFacadeInterface $momentumFacade
+    ) {
+        $this->momentumCallback = $momentumCallback;
+        $this->momentumFacade   = $momentumFacade;
+    }
+
+    /**
+     * Test if this Transport mechanism has starte-d.
+     *
+     * @return bool
+     */
+    public function isStarted()
+    {
+        return $this->started;
+    }
+
+    /**
+     * Start this Transport mechanism.
+     *
+     * @throws \Swift_TransportException
+     */
+    public function start()
+    {
+        $this->started = true;
+    }
+
+    /**
+     * Stop this Transport mechanism.
+     */
+    public function stop()
+    {
```

Comment:
something missing here?

Reply:
I do not think so. There are several methods which are forced by Interface but never used actually. I met same problems when working on SendGrid.